kingpin/actors/rightscale/api.py

```diff
@@ -327,11 +345,11 @@ def update_server_array(self, array, params):
             <updated rightscale array object>
         """
 
-        log.debug('Patching ServerArray (%s) with new params: %s' %
-                  (array.soul['name'], params))
-        array.self.update(params=params)
-        updated_array = array.self.show()
-        return updated_array
+        log.debug('Patching %s with new params: %s' %
+                  (resource.soul['name'], params))
+        resource.self.update(params=params)
+        updated_resource = resource.self.show()
```

Comment:
Not really -- this is a synchronous method wrapped in `@concurrent.run_on_executor`, just like `self.show()` is. The only change I really made here was to make this not array-specific, and let it be re-used for other non-array resources that take the `params` arg.